Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -436,8 +436,13 @@ theorem coe_linearMap : ⇑(Algebra.linearMap R A) = algebraMap R A :=
   rfl
 #align algebra.coe_linear_map Algebra.coe_linearMap
 
-instance id : Algebra R R :=
-  (RingHom.id R).toAlgebra
+/- The identity map inducing an `Algebra` structure. We override the data fields
+  `toFun` and `smul` because `RingHom.id` is not reducible.
+  see library note [reducible non-instances]. -/
+instance id : Algebra R R where
```

Comment:
```suggestion
/-- Every commutative ring is an algebra over itself -/
instance id : Algebra R R where
  /- We override the data fields `toFun` and `smul` because `RingHom.id` is not reducible.
  See note [reducible non-instances] for a related explanation. -/
```
I don't think this belongs in the docstring, it's an implementation detail.